src/round.rs

```diff
@@ -0,0 +1,162 @@
+// This is a part of Chrono.
+// See README.md and LICENSE.txt for details.
+
+use Timelike;
+use std::ops::{Add, Sub};
+use oldtime::Duration;
+
+/// Extension trait for subsecond rounding or truncation to a maximum number
+/// of digits. Rounding can be used to decrease the error variance when
+/// serializing/persisting to lower precision. Truncation is the default
+/// behavior in Chrono display formatting.  Either can be used to guarantee
+/// equality (e.g. for testing) when round-tripping through a lower precision
+/// format.
+pub trait SubSecondRound {
+    /// Return a copy rounded to the specified number of subsecond digits. With
+    /// 9 or more digits, self is returned unmodified. Halfway values are
+    /// rounded up (away from zero).
+    fn round(self, digits: u16) -> Self;
```

Comment:
All internal implementations of `TimeLike` are `Copy` I believe.  I don't think there is any way to avoid a copy in the implementation or otherwise make it more efficient with this. It might actually thwart compiler optimizations?  Presumably that is why the `Add` and `Sub` implementations are also not by reference?  I can be made to work if necessary however. See da4c19b using `Clone`, or 14504b2 using `Copy`.  Let me know if you still think either of these are an improvement, and I'll hope to better understand the rationale.

Reply:
Hmm yeah the tests that you have demonstrate that they work as though they're copy. I wasn't concerned so much about efficiency as I was about a clear and easy-to-use API. So, specifically, if we weren't copy then the following two lines would fail with a use-after-move error:

```
dt.round_subsec(1);
dt.round_subsec(1);
```

But you have a bunch of tests that do that. I thought that it wouldn't work because e.g. [`DateTime<TimeZone>`](https://github.com/chronotope/chrono/blob/2a825edd5c3fc228f834bb445ff409804b9ff539/src/datetime.rs#L49) doesn't `derive(Copy)` so I'm not sure how consistency is being maintained right now.

Reply:
The lack of derive surprised me as well, but its actually implemented manually at [datetime.rs#L437](https://github.com/chronotope/chrono/blob/master/src/datetime.rs#L437) and even [tested](https://github.com/chronotope/chrono/blob/master/src/datetime.rs#L1218).  Note that use-after-move, if lacking `Copy`, also applies to chrono's  `Add` and `Sub`:

~~~rust
let future = dt + Duration::nanoseconds(1);
let past = dt - Duration::nanoseconds(1);
~~~

Reply:
Oh, ha, yup. Okay I'll give this one more once-over and probably merge tonight. Thanks!